internal/tools/aws/dynamodb.go

```diff
@@ -5,29 +5,37 @@
 package aws
 
 import (
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/aws/awserr"
-	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
 	"github.com/pkg/errors"
 	log "github.com/sirupsen/logrus"
 )
 
 // DynamoDBEnsureTableDeleted is used to check if DynamoDB table exists and delete it.
 func (a *Client) DynamoDBEnsureTableDeleted(tableName string, logger log.FieldLogger) error {
 	// First check if table still exists.
-	_, err := a.Service().dynamodb.DescribeTable(&dynamodb.DescribeTableInput{
-		TableName: aws.String(tableName),
-	})
-	if aerr, ok := err.(awserr.Error); ok {
-		if aerr.Code() == dynamodb.ErrCodeResourceNotFoundException {
+	_, err := a.Service().dynamodb.DescribeTable(
+		context.TODO(),
+		&dynamodb.DescribeTableInput{
+			TableName: aws.String(tableName),
+		})
+
+	if err != nil {
+		var awsErr *types.ResourceNotFoundException
+		if errors.As(err, &awsErr) {
```

Comment:
Just curious. What if It is a different type of error?

Reply:
I'm assuming (from the previous logic) that if any error raises here, we will try to delete the database anyway (since that's expected), and any possible error would raise there instead. We could however have some kind of warning here (without ending the function) so is better traceable. What do you think?

Reply:
A warning message will be helpful.